lib/text/ui_text_displayer.js

```diff
@@ -382,15 +392,44 @@ shaka.text.UITextDisplayer = class {
       // TODO: Revisit this as we update the TTML parser.  See special case for
       // backgroundImage + region above.
       const percentageUnit = shaka.text.CueRegion.units.PERCENTAGE;
+
       const heightUnit = cue.region.heightUnits == percentageUnit ? '%' : 'px';
       const widthUnit = cue.region.widthUnits == percentageUnit ? '%' : 'px';
-      const viewportAnchorUnit =
-          cue.region.viewportAnchorUnits == percentageUnit ? '%' : 'px';
       style.height = cue.region.height + heightUnit;
       style.width = cue.region.width + widthUnit;
       style.position = 'absolute';
-      style.top = cue.region.viewportAnchorY + viewportAnchorUnit;
-      style.left = cue.region.viewportAnchorX + viewportAnchorUnit;
+
+      // Handle anchoring. Convert everything to percentages of the video
+      // container to avoid comparing percentages to pixels.
+      let width = cue.region.width;
+      let height = cue.region.height;
+      let viewportAnchorX = cue.region.viewportAnchorX;
+      let viewportAnchorY = cue.region.viewportAnchorY;
+
+      if (cue.region.widthUnits != percentageUnit) {
+        width = width / videoBoundingBox.width * 100;
+      }
+
+      if (cue.region.heightUnits != percentageUnit) {
+        height = height / videoBoundingBox.height * 100;
+      }
+
+      if (cue.region.viewportAnchorUnits != percentageUnit) {
+        viewportAnchorX = viewportAnchorX / videoBoundingBox.width * 100;
+        viewportAnchorY = viewportAnchorY / videoBoundingBox.height * 100;
+      }
+
+
+      // Region and viewport anchors are similar to pinning a paper on a board.
+      // Region anchor is analogous to where the pin goes on the paper, where
+      // the paper is the region. Viewport anchor is analogous to where the
+      // paper goes on the board, where the board is the video.
+
+      // regionAnchorX and regionAnchorY are guaranteed percentages.
+      const pinLocationX = cue.region.regionAnchorX / 100 * width;
+      const pinLocationY = cue.region.regionAnchorY / 100 * height;
+      style.top = cue.region.viewportAnchorY - pinLocationY + '%';
+      style.left = cue.region.viewportAnchorX - pinLocationX + '%';
```

Comment:
You're doing everything in terms of percentages, correct?  This looks like you're converting percentages to px for pinLocationX/Y.

Also, you're not using the computed values of viewportAnchorX/Y which are in percentages.  Instead, you're using the originals, which might be either unit.

So unless I misread, it's possible here that you convert % to px, subtract that from a value that might be either % or px, then append '%' as a unit.